### PR TITLE
FossGraph. Add tables for link projects on FossGraphView

### LINKED ISSUE #1962: FossGraph. Add tables for link projects on FossGraphView
*  add tables for link project
*  add description for vulnerability

### PR DESCRIPTION
**Pull request by Cheshiriks**
* added tables for link project
* added description for vulnerability

Closes #1962

### CODE REVIEW

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt [hunk 1]
```diff
@@ -31,6 +29,7 @@ typealias VulnerabilityDtoList = List<VulnerabilityDto>
 @RequestMapping(path = ["/api/$v1/vulnerabilities"])
 class VulnerabilityController(
     private val vulnerabilityService: VulnerabilityService,
+    private val vulnerabilityDescriptionService: VulnerabilityDescriptionService,
```
Comment (sanyavertolet) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt:32]: Why do you need `VulnerabilityDescriptionService`? I mean `VulnerabilityDescription` is just a pair of `Vulnerability` id and its description. Maybe encapsulate all the `Vulnerability` logic in `VulnerabilityService`

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/FossGraphView.kt [hunk 2]
```diff
@@ -6,83 +6,130 @@
...
 )
 val fossGraph: FC<FossGraphViewProps> = FC { props ->
     useBackground(Style.WHITE)
 
     val params = useParams()
     val vulnerabilityName = params["vulnerabilityName"]!!.toString()
 
-    div {
-        className = ClassName("card card-body mt-0")
+    val (vulnerability, setVulnerability) = useState<VulnerabilityDto?>(null)
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/FossGraphView.kt:58]: I do not like an idea of `vulnerability` being nullable. Maybe some stub?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/FossGraphView.kt [hunk 3]
```diff
@@ -6,83 +6,130 @@
...
+
+        setVulnerabilityProjects(vulnerabilityProjectsNew)
+    }
+
+    val openSourceProjectTable: FC<TableProps<VulnerabilityProjectDto>> = tableComponent(
+        columns = {
+            columns {
+                column(id = "name", header = "Name", { name }) { cellContext ->
+                    Fragment.create {
+                        ReactHTML.td {
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/FossGraphView.kt:90]:
```suggestion
                        td {
```

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt [hunk 4]
```diff
@@ -47,4 +46,19 @@ class VulnerabilityController(
...
+        summary = "Get vulnerability by name.",
+        description = "Get vulnerability by name.",
+    )
+    @ApiResponse(responseCode = "200", description = "Successfully fetched vulnerability by name")
+    fun getVulnerabilityWithDescriptionByName(
+        @RequestParam name: String,
+    ): Mono<VulnerabilityDto> = blockingToMono {
+        val description = vulnerabilityDescriptionService.findByVulnerabilityName(name)?.description
+        vulnerabilityService.findByName(name)?.toDtoWithDescription(description)
+    }
```
[the remark this answers is not in the document]
Comment (nulls) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt:63]: I don't think so, I think all blocking operations should be wrapped by a single `blockingToMono` as much as possible

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt (hunk 1 above)
Comment (Cheshiriks) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt]: Agree. Deleted _VulnerabilityDescriptionService_

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt [hunk 5]
```diff
@@ -47,4 +46,19 @@ class VulnerabilityController(
             Vulnerability::toDto
         )
     }
+
+    @GetMapping("/by-name-with-description")
+    @PreAuthorize("permitAll()")
```
[the remark this answers is not in the document]
Comment (Cheshiriks) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt]: Deleted it

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (Cheshiriks) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt]: Removed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/FossGraphView.kt (hunk 2 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/FossGraphView.kt]: Ok. Added _empty_

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/FossGraphView.kt [hunk 6]
```diff
@@ -6,83 +6,130 @@
...
+        useServerPaging = false,
+        usePageSelection = false,
+    )
+
+    val projectTable: FC<TableProps<VulnerabilityProjectDto>> = tableComponent(
+        columns = {
+            columns {
+                column(id = "name", header = "Name", { name }) { cellContext ->
+                    Fragment.create {
+                        ReactHTML.td {
```
[the remark this answers is not in the document]
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/FossGraphView.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/FossGraphView.kt (hunk 3 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/FossGraphView.kt]: Fixed

**Pull request merged**